### PR TITLE
Netcore/members signin manager [merge after roles]

### PR DESCRIPTION
**Pull request by emmagarland**
### Prerequisites

- [ ] I have added steps to test this contribution in the description below

### Description

New Members SignInmanager. 

Once roles PR #9946 is merged, this PR will reduce in size. It should be merged after #9902 and #9946 

* Moves shared logic for sign in manager to an abstract base class. This class allows us to define custom scheme types if we want, though for members we'll be using the default aspnet identity schemes
* New MemberClaimsPrincipalFactory instead of using the default
* Fixes BackOfficeClaimsPrincipalFactory to use the correct auth type. 
* Uses the correct UmbracoIdentityRole class
* fixes up MemberSignInManagerTests

### CODE REVIEW

**Comment by Shazwazza** [comment]
@emmagarland  I'll get to reviewing #9946 tomorrow since i'll need the work that's being done in here for the SignInManager. Just so I don't overlap with your work, do you think you'll be doing any work on this branch tomorrow (friday) or monday? Else I'm happy to contribute to this on those days, or sub divide the work for the parts I need for member macros.

**Comment by emmagarland** [comment]
> @emmagarland I'll get to reviewing #9946 tomorrow since i'll need the work that's being done in here for the SignInManager. Just so I don't overlap with your work, do you think you'll be doing any work on this branch tomorrow (friday) or monday? Else I'm happy to contribute to this on those days, or sub divide the work for the parts I need for member macros.

@Shazwazza thanks for flagging - I don't think I'll be able to do any work on this Friday or Monday, so the more help the merrier :)

**Review by Shazwazza — COMMENTED** [review]
I've made a couple comments and pushed a bunch of changes. The sign in managers code is now all shared and I fixed up a couple of things. Tests should be passing so I think we can merge. I'll be further testing the manager with the updates to the member macros so we can have end to end tests. ... but holidays start today so will have to wait till I'm back :)

File: src/Umbraco.Web.Common/Constants/Security.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+namespace Umbraco.Cms.Web.Common.Constants
+{
+    public static class Security
+    {
+        //TODO: implement 2Factor external
+
+        public const string MemberAuthenticationType = "UmbracoMember";
```
[the remark this answers is not in the document]
Comment (emmagarland) [reply src/Umbraco.Web.Common/Constants/Security.cs]: Makes sense

**Comment by emmagarland** [comment]
> I've made a couple comments and pushed a bunch of changes. The sign in managers code is now all shared and I fixed up a couple of things. Tests should be passing so I think we can merge. I'll be further testing the manager with the updates to the member macros so we can have end to end tests. ... but holidays start today so will have to wait till I'm back :)

Awesome stuff Shannon, thanks and have reviewed the comments. Enjoy your holidays!

**Comment by Shazwazza** [comment]
Have just re-merged and pushed. The prev build was failing but only due to strange linux database issues. Tests should all pass now and then I'll merge.

**Comment by emmagarland** [comment]
> Have just re-merged and pushed. The prev build was failing but only due to strange linux database issues. Tests should all pass now and then I'll merge.

Fab @Shazwazza 👍

**Pull request merged**